ow_lander/scripts/action_trajectories.py

```diff
@@ -0,0 +1,1033 @@
+        goal_pose.orientation.z = goal_pose.orientation.z
+        goal_pose.orientation.w = goal_pose.orientation.w
+
+        # Ask the planner to generate a plan to the approximate joint values generated
+        # by kinematics builtin IK solver. For more insight on this issue refer to:
+        # https://github.com/nasa/ow_simulator/pull/60
+        if approximate:
+            move_arm.set_joint_value_target(goal_pose, True)
+        else:
+            move_arm.set_pose_target(goal_pose)
+
+        _, plan, _, _ = move_arm.plan()
+
+        if len(plan.joint_trajectory.points) == 0:  # If no plan found, abort
+            return False
+
+        return plan
+
+    def move_to_pre_trench_configuration_dig_circ(self, move_arm, robot, x_start, y_start):
+        """
+        :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+        :type robot: class 'moveit_commander.RobotCommander'
+        :type x_start: float
+        :type y_start: float
+        """
+        # Initilize to current position
+        joint_goal = move_arm.get_current_pose().pose
+        robot_state = robot.get_current_state()
+        move_arm.set_start_state(robot_state)
+
+        # Compute shoulder yaw angle to trench
+        alpha = math.atan2(y_start-constants.Y_SHOU, x_start-constants.X_SHOU)
+        h = math.sqrt(pow(y_start-constants.Y_SHOU, 2) +
+                      pow(x_start-constants.X_SHOU, 2))
+        l = constants.Y_SHOU - constants.HAND_Y_OFFSET
+        beta = math.asin(l/h)
+        # Move to pre trench position, align shoulder yaw
+        joint_goal = move_arm.get_current_joint_values()
+        joint_goal[constants.J_DIST_PITCH] = 0.0
+        joint_goal[constants.J_HAND_YAW] = 0.0
+        joint_goal[constants.J_PROX_PITCH] = -math.pi/2
+        joint_goal[constants.J_SHOU_PITCH] = math.pi/2
+        joint_goal[constants.J_SHOU_YAW] = alpha + beta
+
+        # If out of joint range, abort
+        if (is_shou_yaw_goal_in_range(joint_goal) == False):
+            return False
+
+        joint_goal[constants.J_SCOOP_YAW] = 0
+
+        move_arm.set_joint_value_target(joint_goal)
+        _, plan, _, _ = move_arm.plan()
+        return plan
+
+    def dig_circular(self, move_arm, move_limbs, robot, moveit_fk, args, server_stop):
+        """
+        :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+        :type robot: class 'moveit_commander.RobotCommander'
+        :type moveit_fk: class moveit_msgs/GetPositionFK
+        :type args: List[bool, float, int, float, float, float]
+        """
+
+        circ_traj = None
+        circ_traj = RobotTrajectory()
+
+        x_start = args.x_start
+        y_start = args.y_start
+        depth = args.depth
+        parallel = args.parallel
+        ground_position = args.ground_position
+
+        if not parallel:
+
+            server_stop.get_state()
+            if server_stop.stopped is True:
+                return False
```

Comment:
These three lines are repeated many times.  Three suggestions:

1. Factor this out into a boolean-valued utility function.
2. Issue an Info message when the operation is stopped, otherwise hard to distinguish from normal completion in the output.
3. Minor, and I'm not a Python expert: I think `==` is more appropriate than `is' here, and `not ...` reads even better.

Reply:
Actually it would be totally fine (and completely readable) to just use:
```suggestion
            server_stop.get_state()
            if server_stop.stopped:
                return False
```

Reply:
I agree the repetition of this statement is not ideal, but I think I have a better way to address it, which I think would be better covered in another PR. It involves some changes to arm_action_servers.py, so I would be happy to implement it myself as part of my refactor ticket, https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-904. My proposal is as follows... 

As best as I can tell the `server_stop.stopped` check has to occur before any call to `move_arm.plan()`. So rather than calling a function each time before the call to `plan()`, I suggest we wrap the `plan()` function in an `*ActionServer` version of the `plan()` function that does two things: **1)** Checks the state of `server_stop` and throws and exception if it is stopped, and **2)** calls the `MoveGroupCommander`'s version of `plan()` returning its return values. Then all of these large blocks of movement code simply have to be put inside of a try-catch block, and the case where stop is called is handled inside the catch.

Reply:
I agree the repetition of this statement is not ideal, but I think I have a better way to address it, which I think would be better covered in another PR. It involves some changes to arm_action_servers.py, so I would be happy to implement it myself as part of my refactor ticket, https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-904. My proposal is as follows... 

As best as I can tell the `server_stop.stopped` check has to occur before any call to `move_arm.plan()`. So rather than calling a function each time before the call to `plan()`, I suggest we wrap the `plan()` function in an `*ActionServer` version of the `plan()` function that does two things: **1)** Checks the state of `server_stop` and throws and exception if it is stopped, and **2)** calls the `MoveGroupCommander`'s version of `plan()` returning its return values. Then all of these large blocks of movement code simply have to be put inside of a try-catch block, and the case where stop is called is handled inside the catch.

Reply:
made the requested change